consensus/types/src/subnet_id.rs

```diff
@@ -149,3 +159,80 @@ impl AsRef<str> for SubnetId {
         subnet_id_to_string(self.0)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
```

Comment:
Not sure how hard it is to craft an easy test case, but I think it would be beneficial to test the validity of `valid_until_epoch`. This means testing that the range [`valid_until_epoch`-`subscription_duration`, `valid_until_epoch`) produce the same subnets as the expected subnets of the tested epoch

Reply:
I have a few tests, where i tested against the pyspec. It's not in the rust code, but what I did was, create a for-loop in python and just go through every epoch and return the epoch when the pyspec tells us that we should be on a new spec. The test compares this fixed python result, with the calculated version in rust.

Reply:
Ok I did the tests in the range `[epoch, valid_until_epoch)` checking the valid_epoch doesn't change and the subnets don't change either. Also tested that in `valid_during_epoch` the results change. I'm therefore convinced this is legit